code-challenges.test.js

```diff
@@ -17,45 +17,112 @@
+    // iterate through the array accessing each element -- maybe able to use .filter()
+    // humm.. i wonder if JS has a .include? method (thinking in Ruby right now...) 
+        //! and JS does have an .includes() method. Also added an OR condition inside the filter to check for capital letter A
+    // ... this way we can use the method to check if the word includes the letter a or A
+    // and return an array with only the words that have a letter a or A
 
+const countainLetterA = (arr) => {
+    return arr.filter((item) => item.includes('a') || item.includes('A'))
+}
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with an expect statement using the variable provided.
 
-var hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-var hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-var hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-
-
-
+describe("isFullHouse", () => {
+    test("check if the array is a full house and returns a boolean", () => {
+        let hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        let hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        let hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
+        expect(isFullHouse(hand1)).toEqual(true)
+        expect(isFullHouse(hand2)).toEqual(false)
+        expect(isFullHouse(hand3)).toEqual(false)
+    })
+  })
+  
 // b) Create the function that makes the test pass.
+// Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+// --> Pseudo Code
+    // create a function that takes in an array as an argument
+    // iterate through the array and check each element - use the forEach() higher order function
+
+    ///// initial approach....
+    // use a counter to check the number of instances of each element
+    // use the value from the counter (ex: counter1 = 3, and counter2 = 2) to check if we have a full house! 
+
+    ///// ... i found this snipet online in which we are declaring a variable counts and assigning an empty object. we apply the forEach method to iterate through the array and for every item in the array we store the item as a key in the counts object and we count the occurances of the item and store as the value of the key. 
+    // now I can check for the key values in the counts object (Object.values(counts)) --> this returns an array. 
+    // we can iterate through the values array (numOfsameItemsArr) for the presence of an element with the value of 2 and the value of 3!!!
+    // - instead of iterating, I used the bulit in method .includes to check for the presence of the elements 2 and 3 in the array.
+    // return true is the array includes 2 and includes 3. else return false
```

Comment:
Great explanation! I've seen a few people use this snippet so it's nice to see an explanation of it!